docs/user/metrics/counter.md

```diff
@@ -1,14 +1,47 @@
 # Counter
 
-Count how often something happens.
+Used to count how often something happens, say how often a certain button was
+pressed. A counter always starts from `0`. Each time you record to a counter,
+its value is incremented.
 
-## Methods
+## Configuration
 
-* `add(amount)` - Increase the counter by `amount`.
+Say you're adding a new counter for how often the refresh button is pressed.
+First you need to add an entry for the counter to the `metrics.yaml` file:
+
+```YAML
+controls:
+  refresh_pressed:
+    type: counter
+    description: >
+      Counts how often the refresh button is pressed.
+    ...
+```
+
+## API
+
+```Kotlin
+import org.mozilla.yourApplication.GleanMetrics.Controls
+
+Controls.refreshPressed.add() // Adds 1 to the counter.
+Controls.refreshPressed.add(5) // Adds 5 to the counter.
+```
+
+There are test APIs available too:
+
+```Kotlin
+import org.mozilla.yourApplication.GleanMetrics.Controls
+Glean.enableTestingMode()
+
+// Was anything recorded?
+assertTrue(Controls.refreshPressed.testHasValue())
+// Does the counter have the expected value?
+assertEquals(6, Controls.refreshPressed.testGetValue())
+```
 
 ## Limits
 
-* Only increments, saturates at the limits of a 32-bit unsigned integer
+* Only increments, saturates at the limits of a 32-bit signed integer
```

Comment:
Mh, was this misdocumented before or did we change the behaviour here?

Reply:
Not documented for Glean AC at all, and misdocumented here (probably because I first implemented it using unsigned integers).

Now that I see this here: we actually don't do an explicit check at the limit. We might want to add that?

Reply:
https://bugzilla.mozilla.org/show_bug.cgi?id=1554116

Reply:
https://bugzilla.mozilla.org/show_bug.cgi?id=1556496